reverse_proxy_handler.cpp

```diff
@@ -237,3 +200,62 @@ std::pair<std::string, std::string> ReverseProxyHandler::ProcessHeaderLine(std::
 
   return std::make_pair( header_field, header_value);
 }
+
+bool ReverseProxyHandler::ParseRedirect(std::istream& response_stream, std::string& redirect_URI,
+                                        std::string& redirect_host)
+{
+  std::string header;
+  while (std::getline(response_stream, header) && header != "\r")
+  {
+    // Find the new location header
+    if (!header.empty())
+    {
+      std::pair<std::string, std::string> parsed_header = ProcessHeaderLine(header);
+      if (parsed_header.first == "Location")
```

Comment:
Maybe should we change it all to lowercase and check against "location"? Just in case some headers have different capitalizations..?